webviz_config/templates/webviz_template.py.jinja2

```diff
@@ -113,6 +113,21 @@ else:
             ])]),
         html.Div(className="pageContent", id="page-content")])
 
+plugins = []
+{% for page in pages %}
+{% for content in page.content %}
+{% if content is not string %}
+plugins.append(webviz_config.plugins.{{ content._call_signature[0] }})
+{% endif %}
+{% endfor %}
+{% endfor %}
+use_oauth2 = any(plugin.oauth2 for plugin in plugins if hasattr(plugin, "oauth2"))
+oauth2 = None
+if use_oauth2:
+    oauth2 = webviz_config.Oauth2(app.server)
+    oauth2.set_oauth2_endpoints()
+    if __name__ != "__main__":
+        oauth2.set_oauth2_before_request_decorator()
```

Comment:
```suggestion
oauth2 = webviz_config.Oauth2(app.server) if use_oauth2 else None
```
given that `set_oauth2_endpoints()` becomes part of `Ouaht2.__init__` (does that make sense, i.e. do you want to always call it on initizalition?) and that 
```python
oauth2.set_oauth2_before_request_decorator()
```
is moved to an `else:` block after the already existing `if __name__ == "__main__"` block.